<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>FeatheredEdgeDialog</class>
 <widget class="QDialog" name="FeatheredEdgeDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>400</width>
    <height>380</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Feathered Edge</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QGroupBox" name="originGroupBox">
     <property name="title">
      <string>Origin Point</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <widget class="QLabel" name="instructionLabel">
        <property name="text">
         <string>Select the origin point for the fall-off effect:</string>
        </property>
        <property name="wordWrap">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <widget class="QPushButton" name="pickPointButton">
          <property name="text">
           <string>Pick Point</string>
          </property>
          <property name="toolTip">
           <string>Click to pick a point from the cloud (currently not implemented - use manual input)</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="useCloudCenterButton">
          <property name="text">
           <string>Use Cloud Center</string>
          </property>
          <property name="toolTip">
           <string>Use the bounding box center of the cloud as origin</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QGridLayout" name="gridLayout">
        <item row="0" column="0">
         <widget class="QLabel" name="labelX">
          <property name="text">
           <string>X:</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QDoubleSpinBox" name="originXSpinBox">
          <property name="decimals">
           <number>6</number>
          </property>
          <property name="minimum">
           <double>-999999.000000000000000</double>
          </property>
          <property name="maximum">
           <double>999999.000000000000000</double>
          </property>
          <property name="toolTip">
           <string>X coordinate of origin point</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="labelY">
          <property name="text">
           <string>Y:</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QDoubleSpinBox" name="originYSpinBox">
          <property name="decimals">
           <number>6</number>
          </property>
          <property name="minimum">
           <double>-999999.000000000000000</double>
          </property>
          <property name="maximum">
           <double>999999.000000000000000</double>
          </property>
          <property name="toolTip">
           <string>Y coordinate of origin point</string>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="labelZ">
          <property name="text">
           <string>Z:</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QDoubleSpinBox" name="originZSpinBox">
          <property name="decimals">
           <number>6</number>
          </property>
          <property name="minimum">
           <double>-999999.000000000000000</double>
          </property>
          <property name="maximum">
           <double>999999.000000000000000</double>
          </property>
          <property name="toolTip">
           <string>Z coordinate of origin point</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="parametersGroupBox">
     <property name="title">
      <string>Fall-off Parameters</string>
     </property>
     <layout class="QFormLayout" name="formLayout">
      <item row="0" column="0">
       <widget class="QLabel" name="maxDistanceLabel">
        <property name="text">
         <string>Max Distance:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QDoubleSpinBox" name="maxDistanceSpinBox">
        <property name="decimals">
         <number>3</number>
        </property>
        <property name="minimum">
         <double>0.001000000000000</double>
        </property>
        <property name="maximum">
         <double>999999.000000000000000</double>
        </property>
        <property name="value">
         <double>10.000000000000000</double>
        </property>
        <property name="toolTip">
         <string>Maximum distance for fall-off effect. Points beyond this distance will be removed.</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="falloffTypeLabel">
        <property name="text">
         <string>Fall-off Type:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QComboBox" name="falloffTypeComboBox">
        <property name="toolTip">
         <string>Type of fall-off curve to apply</string>
        </property>
        <item>
         <property name="text">
          <string>Linear</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Exponential</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Gaussian</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Smoothstep</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="strengthLabel">
        <property name="text">
         <string>Strength:</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QDoubleSpinBox" name="strengthSpinBox">
        <property name="decimals">
         <number>2</number>
        </property>
        <property name="minimum">
         <double>0.100000000000000</double>
        </property>
        <property name="maximum">
         <double>10.000000000000000</double>
        </property>
        <property name="singleStep">
         <double>0.100000000000000</double>
        </property>
        <property name="value">
         <double>1.000000000000000</double>
        </property>
        <property name="toolTip">
         <string>Fall-off strength. Higher values create steeper fall-off.</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="randomSeedLabel">
        <property name="text">
         <string>Random Seed:</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QSpinBox" name="randomSeedSpinBox">
        <property name="minimum">
         <number>0</number>
        </property>
        <property name="maximum">
         <number>2147483647</number>
        </property>
        <property name="value">
         <number>0</number>
        </property>
        <property name="toolTip">
         <string>Random seed for reproducibility (0 = use random seed)</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>FeatheredEdgeDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>FeatheredEdgeDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
